Objects/typeobject.c

```diff
@@ -3256,6 +3256,9 @@ type_new_get_bases(type_new_ctx *ctx, PyObject **type)
         if (winner->tp_new != type_new) {
             /* Pass it to the winner */
             *type = winner->tp_new(winner, ctx->args, ctx->kwds);
+            if (*type == NULL) {
+                return -1;
+            }
             return 1;
```

Comment:
Small nit, feel free to ignore

```suggestion
            return *type == NULL ? -1 : 1;
```

Reply:
I prefer to "handle errors explicitly". Well, it's just a personal coding style preference.